charmhelpers/fetch/ubuntu.py

```diff
@@ -189,17 +190,185 @@
 CMD_RETRY_COUNT = 3  # Retry a failing fatal command X times.
 
 
+def dpkg_list(packages):
+    """Get data from system dpkg database for package.
+
+    Note that one of the main purposes for this function is to avoid dependency
+    on the ``python-apt`` package.
+
+    The indicated package is a wrapper around the ``apt`` C++ library which is
+    tightly connected to the version of the distribution it is shipped on.
+    This in turn makes it incredibly hard to distribute as a wheel for a charm
+    that supports a large span of distro versions.
+
+    :param packages: Packages to get data from
+    :type packages: list[str]
+    :returns: Structured data about installed packages, keys like
+              ``dpkg-query --list``
+    :rtype: dict
```

Comment:
This should be `Dict[str, str]` I believe

Reply:
:+1: